README.markdown

```diff
@@ -1,144 +1,167 @@
+```html
+<script>
+  UPLOADCARE_PUBLIC_KEY = 'YOUR_PUBLIC_KEY';
+</script>
+```
+
+Your secret key is not required for the widget
+(it’s quite careless for your page to include any
+secret keys anyway.)
+
+**Insert widget element** into your form,
 
-After open command line, go to this folder and run:
+```html
+<input type="hidden" role="uploadcare-uploader" name="my_file" />
+```
 
-    vagrant up
-    vagrant ssh
+By default, the library looks for inputs with the specified
+`role` attribute and places widgets there.
+Once a file is uploaded, this `<input>` gets a
+CDN link with a file UUID. Your server then
+receives this link instead of file content.
 
-#### Local environment (without Vagrant)
+We suggest placing the widget somewhere at the top of your form.
+Unlike regular inputs, our widget starts uploading files **immediately**
+after they get selected by a user, not on form submission.
+That way users can fill out the rest of your form while an
+upload is in progress. This can be a real time saver.
 
-Inside folder run
+### Usage with React
 
-    bundle install
-    cd ./test/dummy
-    bundle install
+```javascript
+import React, {Component} from 'react'
+import uploadcare from 'uploadcare-widget'
 
-### Build
+class Uploader extends Component {
 
-* `bundle exec rake js:latest:build` to build assets
-  to the **pkg/latest** directory (with the “latest” suffix).
-* `bundle exec rake js:release:build` to build assets
-  to the **pkg/version** folder (with the current version suffix).
-  The version is specified in `lib/uploadcare-widget/version.rb`.
+  componentDidMount() {
+    uploadcare.start({publicKey: 'YOUR_PUBLIC_KEY'})
+  }
 
-### Development
+  render() {
+    return (
+      <div>
+        <input type='text' role='uploadcare-uploader'/>
+      </div>
+    )
+  }
+}
 
-Go to `test/dummy/`. There is a simple Rails app. Run it:
+export default Uploader
+```
 
-    bundle exec rails server
-    
-Open http://127.0.0.1:3000/. Follow any link. 
-There's going to be a widget or three. Edit code and reload page :-)
+See [full demo](https://github.com/uploadcare/uploadcare-widget-react-demo/).
 
+### Usage with Angular
 
-### Testing
+We have Angular 1 directive for Uploadcare Widget.
+See [angular-uploadcare](https://github.com/uploadcare/angular-uploadcare).
 
-[Jasminerice](https://github.com/bradphelan/jasminerice) 
-installed under the `test/dummy/` Rails app.
+For Angular 2 maybe [this demo](https://plnkr.co/edit/6caWQ6cct4L3715LehxZ?p=preview) can be useful.
 
-To run tests in your browser go to http://127.0.0.1:3000/jasmine.
+## Configuration
 
-For more information see 
-[jasminerice docs](https://github.com/bradphelan/jasminerice).
+The widget is highly customizable through widget options.
+All configuration options and ways to set options
```

Comment:
All configuration options `together | along` with ways to set them are described in `these | the | our` docs.